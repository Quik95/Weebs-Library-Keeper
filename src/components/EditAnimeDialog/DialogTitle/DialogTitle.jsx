import React, { memo } from 'react';
import PropTypes from 'prop-types';

// MUI
import MuiDialogTitle from '@material-ui/core/DialogTitle';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';

// styles
import useStyles from '../EditAnimeDialog.styles';

const DialogTitle = memo(function DialogTitle({ onClose, children, ...other }) {
  const classes = useStyles();

  return (
    <MuiDialogTitle
      disableTypography
      className={classes.dialogTitle}
      {...other}
    >
      <Typography variant="h6">{children}</Typography>
      {onClose ? (
        <IconButton
          aria-label="close"
          className={classes.closeButton}
          onClick={onClose}
        >
          <CloseIcon />
        </IconButton>
      ) : null}
    </MuiDialogTitle>
  );
});

DialogTitle.propTypes = {
  onClose: PropTypes.func.isRequired,
  children: PropTypes.string.isRequired,
};

export default DialogTitle;
