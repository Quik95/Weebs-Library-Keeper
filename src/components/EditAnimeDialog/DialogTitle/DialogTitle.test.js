import DialogTitle from './DialogTitle';

describe('', () => {
  const handleCloseMock = jest.fn(() => {});

  const wrapper = shallow(<DialogTitle onClose={handleCloseMock} />);

  it('Renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
