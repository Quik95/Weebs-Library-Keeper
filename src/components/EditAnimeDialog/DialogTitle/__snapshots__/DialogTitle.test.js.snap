// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<WithStyles(ForwardRef(DialogTitle))
  disableTypography={true}
>
  <WithStyles(ForwardRef(Typography))
    variant="h6"
  />
  <WithStyles(ForwardRef(IconButton))
    aria-label="close"
    onClick={[MockFunction]}
  >
    <CloseIcon />
  </WithStyles(ForwardRef(IconButton))>
</WithStyles(ForwardRef(DialogTitle))>
`;
